spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/TcpSendingMessageHandler - Copy.txt

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.tcp;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledFuture;
+
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.integration.handler.AbstractMessageHandler;
+import org.springframework.integration.ip.IpHeaders;
+import org.springframework.integration.ip.tcp.connection.AbstractClientConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.AbstractConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.ClientModeCapable;
+import org.springframework.integration.ip.tcp.connection.ClientModeConnectionManager;
+import org.springframework.integration.ip.tcp.connection.ConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.TcpConnection;
+import org.springframework.integration.ip.tcp.connection.TcpConnectionFailedCorrelationEvent;
+import org.springframework.integration.ip.tcp.connection.TcpSender;
+import org.springframework.integration.support.management.ManageableLifecycle;
+import org.springframework.integration.support.utils.IntegrationUtils;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandlingException;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.util.Assert;
+
+/**
+ * Tcp outbound channel adapter using a TcpConnection to
+ * send data - if the connection factory is a server
+ * factory, the TcpListener owns the connections. If it is
+ * a client factory, this object owns the connection.
+ *
+ * @author Gary Russell
+ * @author Artem Bilan
+ * @since 2.0
+ */
+public class TcpSendingMessageHandler extends AbstractMessageHandler implements
```

Comment:
What is this file for?

Reply:
sorry it is not required at all. my mistake.